File: Application/DTOs/UserDto.cs
using Application.Common;
using Domain.Entities;
using Microsoft.AspNetCore.Identity;

namespace Application.DTOs;

public class UserDto : BaseDto<int>
{
    public string AspNetUserId { get; set; } = null!;
    public string FirstName { get; set; } = null!;
    public string LastName { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string? PhoneNumber { get; set; }
    public string? RefreshToken { get; set; }
    public DateTime? RefreshTokenExpiryTime { get; set; }
    public bool IsBlocked { get; set; }
    public string? Role { get; set; }
    public virtual IdentityUser? AspNetUser { get; set; } = null!;
    public virtual EmployeeDto? Employee { get; set; } = null!;
    public virtual ManagerDto? Manager { get; set; } = null!;
    public virtual UserPhotoDto? UserPhoto { get; set; } = null!;
    public virtual ICollection<NotificationGroupDto> NotificationGroups { get; set; } = new List<NotificationGroupDto>();
    public virtual ICollection<Notification> Notifications { get; set; } = new List<Notification>();
}

File: UI/src/app/services/dtos/user.dto.ts
import {ManagerDto} from "./manager.dto";
import {EmployeeDto} from "./employee.dto";
import {UserPhotoDto} from "./user-photo.dto";

export interface UserDto {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  username: string;
  isBlocked: boolean;
  manager: ManagerDto | null;
  employee: EmployeeDto | null;
  userPhoto: UserPhotoDto | null;
  role: string;
}
